### PR TITLE
Implement idle timeout in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR implements xxx.Connection.IdleTimeout in C#.

It also includes the same tests as in C++.

### CODE REVIEW

**Review by pepone — APPROVED** [review]
Looks good. Just a few comments.

File: csharp/test/Ice/idleTimeout/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*.html
+*.xml
+*.xaml
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/test/Ice/idleTimeout/.gitignore:3]: Not sure. We have this .gitignore is all or most Ice tests. We should check in a follow-up PR which .gitignore if any we need for the tests.

File: csharp/src/Ice/Properties.cs [hunk 2]
```diff
@@ -500,6 +500,12 @@ public Properties ice_clone_()
         return clonedProperties;
     }
 
+    /// <summary>
+    /// Alias for <see cref="Clone" />. Provided for source compatibility with Ice 3.7 and earlier versions.
+    /// </summary>
+    /// <returns>A copy of this property set.</returns>
+    public Properties ice_clone_() => Clone();
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/src/Ice/Properties.cs:507]: Fixed.

File: csharp/test/Ice/idleTimeout/AllTests.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        TextWriter output)
+    {
+        output.Write("testing that the idle check does not abort a connection that receives heartbeats.. ");
+        output.Flush();
+
+        await p.ice_pingAsync();
+        Connection? connection = p.ice_getCachedConnection();
+        test(connection is not null);
+
+        await p.sleepAsync(2000); // the implementation in the server sleeps for 20,000ms
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/test/Ice/idleTimeout/AllTests.cs:42]: Fixed.

File: csharp/test/Ice/idleTimeout/AllTests.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    private static async Task testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(
+        Test.TestIntfPrx p,
+        TextWriter output)
+    {
+        output.Write("testing that the idle check does not abort a connection that receives heartbeats.. ");
+        output.Flush();
+
+        await p.ice_pingAsync();
+        Connection? connection = p.ice_getCachedConnection();
+        test(connection is not null);
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/test/Ice/idleTimeout/AllTests.cs:40]: The ping is really the Arrange part of this test. The test executes on an established connection. We want to take the connection establishment - and how long it takes - out of it.

Removed line 39 & 40 that are surperfluous.

File: csharp/test/Ice/idleTimeout/AllTests.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        output.Flush();
+
+        // Create a new communicator with the desired properties.
+        properties = properties.Clone();
+        properties.setProperty("Ice.Connection.IdleTimeout", "3");
+        properties.setProperty("Ice.Warn.Connections", "0");
+        Communicator communicator = Util.initialize(new InitializationData { properties = properties });
+        Test.TestIntfPrx p = Test.TestIntfPrxHelper.createProxy(communicator, proxyString);
+        Connection connection = p.ice_getConnection();
+        test(connection is not null);
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/test/Ice/idleTimeout/AllTests.cs:65]: It's like the previous test. We don't want to include the connection establishment - and how long that takes - in the Act part of the test.

**Pull request merged**